osc-server/src/test/java/org/osc/core/broker/service/servicefunctionchain/UpdateServiceFunctionChainServiceTest.java

```diff
@@ -60,91 +60,77 @@ public void testDispatch_WithNullRequest_ThrowsNullPointerException() throws Exc
 	@Test
 	public void testDispatch_WithNonEmptyListOfVsIdsUpdateValidationSuccessful_ValidationSucceeds() throws Exception {
 		
+		//Arrange
 		BaseDto dto = new BaseDto();
 		dto.setId(this.sfc.getId());
 		dto.setParentId(this.vc.getId());
 		request.setDto(dto);
 		request.setName(this.sfc.getName());
 		List<Long> vsIds = new ArrayList<Long>();
+		
 		vsIds.add(this.vs.getId());
 		vsIds.add(this.vs1.getId());
 		request.setVirtualSystemIds(vsIds);
 		Mockito.when(this.service.validator.validateAndLoad(request)).thenReturn(this.sfc);
+		
 		// Act.
 		BaseJobResponse response = this.service.dispatch(request);
+		
+		//Assert
 		Assert.assertNotNull("The returned response should not be null.", response);
 	}
 	
 	@Test
 	public void testDispatch_WhenVsIdsListUpdateValidationSuccessful_ValidationSucceeds() throws Exception {
 		
+		//Arrange
 		BaseDto dto = new BaseDto();
 		dto.setId(this.sfc.getId());
 		dto.setParentId(this.vc.getId());
 		request.setDto(dto);
 		request.setName(this.sfc.getName());
 		List<Long> vsIds = new ArrayList<Long>();
+		
 		vsIds.add(this.vs.getId());
 		vsIds.add(this.vs1.getId());
 		request.setVirtualSystemIds(vsIds);
 		Mockito.when(this.service.validator.validateAndLoad(request)).thenReturn(this.sfc);
-		// Act.
-		BaseJobResponse response = this.service.dispatch(request);
-		Assert.assertNotNull("The returned response should not be null.", response);
-	}
-	
-	@Test
-	public void testDispatch_sfcVirutalSystemIdsInOrder_ValidationSucceeds() throws Exception {
 		
-		BaseDto dto = new BaseDto();
-		dto.setId(this.sfc.getId());
-		dto.setParentId(this.vc.getId());
-		request.setDto(dto);
-		request.setName(this.sfc.getName());
-		List<Long> vsIds = new ArrayList<Long>();
-		vsIds.add(this.vs1.getId());
-		vsIds.add(this.vs.getId());
-		request.setVirtualSystemIds(vsIds);
-		Mockito.when(this.service.validator.validateAndLoad(request)).thenReturn(this.sfc);
 		// Act.
-		//update sfc with virtual system ids list {2,1}
 		BaseJobResponse response = this.service.dispatch(request);
 		
-		//check Virtual system is as updated
-		List<Long> sfcVsIdList = this.em.find(ServiceFunctionChain.class, this.sfc.getId()).
-				getVirtualSystems().stream().map(vss -> vss.getId()).collect(Collectors.toList());
-		//read back sfc and cross check with local Virtual system list for order
-		Assert.assertEquals(vsIds, sfcVsIdList);
+		//Assert
 		Assert.assertNotNull("The returned response should not be null.", response);
```

Comment:
Again, assert what should be in the response. Also how do we know that the service is calling the db with the correct info? If there is a regression on that code this is not going to catch it.

Reply:
added response id check